### PR TITLE
[PAY-1437] Fix drawers sharing state causing crash

### PR DESCRIPTION
**Pull request by dharit-tan**
### Description
Bug: if two different native drawers were opened/closed rapidly, app would crash. This is because the first drawer would set the shared `data`, then when it was closed it would be marked as 'closing' but wouldn't actually unset `data` until its close animation completed. If the 2nd drawer was opened before the animation completed, it would open and read `data`, but then `data` would get set to `null` by the 1st drawer.
Solution: make each drawer have its own `data` in a map.

### Dragons

Touches a bunch of drawers. I've tested all and have asked AJ to test WalletConnect for next release.

### How Has This Been Tested?

Local ios stage

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/store/drawers/selectors.ts [hunk 1]
```diff
@@ -2,8 +2,10 @@ import type { AppState } from '../'
 
 import type { Drawer, DrawerData } from './slice'
 
-export const getData = <TDrawer extends keyof DrawerData>(state: AppState) =>
-  state.drawers.data as DrawerData[TDrawer]
+export const getData = <TDrawer extends keyof DrawerData>(
```
Comment (rickyrombo) [inline packages/mobile/src/store/drawers/selectors.ts:5]: nit: I think typically selectors do `state` as the first argument always?

Although I see below we wrap the selector in a new fn so 🤷 idk what's the best strategy here

**Review by rickyrombo — APPROVED** [review]
Nice find and fix!

File: packages/mobile/src/store/drawers/selectors.ts (hunk 1 above)
Comment (dharit-tan) [reply packages/mobile/src/store/drawers/selectors.ts:5]: yeah was also a bit confused by the next selector below.

**Review by dylanjeffers — APPROVED** [review]
wonderful improvement reed

**Pull request merged**